module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -604,13 +611,54 @@ public function getHolding($bibId, array $patron = null, array $options = [])
                     $item->itemLevelCallNumberPrefix ?? '',
                     $item->itemLevelCallNumber ?? ''
                 );
+
+                $dueDateValue = '';
+                if ($item->status->name == 'Checked out'
+                    && $showDueDate
+                    && $dueDateItemCount < $maxNumDueDateItems
+                ) {
+                    // extract itemId & construct query
+                    $query = [
+                        'query' => 'itemId==' . $item->id
+                    ];
+                    // call /circulation/loans to extract dueDate
+                    foreach ($this->getPagedResults(
+                        'loans',
+                        '/circulation/loans',
+                        $query
+                    ) as $loan) {
+                        if (!isset($loan->returnDate)) {
+                            $dueDate = new DateTime(
+                                $loan->dueDate,
+                                new DateTimeZone('UTC')
+                            );
+                            $localTimezone = (new DateTime)->getTimezone();
+                            $dueDate->setTimezone($localTimezone);
+                            $dueDateValue = $this->dateConverter
+                                ->convertToDisplayDate(
+                                    'U',
+                                    $dueDate->format('U')
+                                );
+                            if ($showTime) {
+                                $dueDateValue = $this->dateConverter
+                                    ->convertToDisplayDateAndTime(
+                                        'U',
+                                        $dueDate->format('U')
+                                    );
+                            }
+                        }
+                    }
+                    $dueDateItemCount += 1;
```

Comment:
You could make this even shorter with `$dueDateItemCount++;`.

Reply:
You are totally right, I do a lot of programming in Swift these days, where the ++ increment was never implemented, as it was deemed ambiguous. So I unlearned a good thing :( I'll update that